osu.Game/Scoring/ScoreInfoExtensions.cs

```diff
@@ -26,6 +27,35 @@ public static IEnumerable<ScoreInfo> OrderByTotalScore(this IEnumerable<ScoreInf
                      // Local scores may not have an online ID. Fall back to date in these cases.
                      .ThenBy(s => s.Date);
 
+        /// <summary>
+        /// Orders an array of <see cref="ScoreInfo"/>s by the selected <see cref="RankingsSort"/>.
+        /// </summary>
+        /// <param name="scores">The array of <see cref="ScoreInfo"/>s to reorder.</param>
+        /// <param name="rankingSort">The attribute to sort the scores by.</param>
+        /// <returns>The given <paramref name="scores"/> ordered by the selected mode.</returns>
+        public static IEnumerable<ScoreInfo> OrderByCriteria(this IEnumerable<ScoreInfo> scores, RankingsSort rankingSort)
+        {
+            switch (rankingSort)
+            {
+                case RankingsSort.Score:
+                    return scores.OrderByDescending(s => s.TotalScore);
+
+                case RankingsSort.Accuracy:
+                    return scores.OrderByDescending(s => s.Accuracy).ThenByDescending(s => s.TotalScore);
+
+                case RankingsSort.Combo:
+                    return scores.OrderByDescending(s => s.MaxCombo).ThenByDescending(s => s.TotalScore);
+
+                case RankingsSort.Misses:
+                    return scores.OrderBy(s => s.Statistics.GetValueOrDefault(HitResult.Miss, 0)).ThenByDescending(s => s.TotalScore);
```

Comment:
This one is also interesting because only taking full misses into account may not be what players want. See: tick misses, especially in contexts like catch where they're droplet misses, etc.

Dunno. Maybe fine for a start, but food for thought.